src/frontend/src/handler/create_table.rs

```diff
@@ -1627,6 +1627,22 @@ pub async fn create_iceberg_engine_table(
                 with_common.insert("connector".to_owned(), "iceberg".to_owned());
                 with_common.insert("database.name".to_owned(), iceberg_database_name.to_owned());
                 with_common.insert("table.name".to_owned(), iceberg_table_name.to_owned());
+
+                if let Some(s) = params.properties.get("hosted_catalog") {
+                    if s.eq_ignore_ascii_case("true") {
+                        with_common.insert("catalog.type".to_owned(), "jdbc".to_owned());
+                        with_common.insert("catalog.uri".to_owned(), catalog_uri.to_owned());
```

Comment:
Maybe we can delete the `#[allow(dead_code)]` branch above now? 🤔

Reply:
Yes. Let me remove it.